test_scripts/AppIconsFolder/ATF_P_AppIconsFolder_exists_no_ReadWritePermissions.lua

```diff
@@ -0,0 +1,143 @@
+    self.applications[RAIParameters.appName] = data.params.application.appID
+  end)
+  self.mobileSession:ExpectResponse(corIdRAI, { success = true, resultCode = "SUCCESS" })
+end
+
+local function pathToAppFolderFunction(appID)
+  return folderNoPermissions .. "/" .. appID .. "_" .. config.deviceMAC .. "/"
+end
+
+--[[ Preconditions ]]
+commonFunctions:newTestCasesGroup("Preconditions")
+
+function Test:Precondition_connectMobile()
+  self:connectMobile()
+end 
+
+function Test.Precondition_setNoPermissionsForIconsFolder()
+  local changePermissions  = assert(os.execute( "chmod 000 " .. folderNoPermissions))
+  if changePermissions ~= true then
+    commonFunctions:userPrint(31, "Permissions for FolderWithoutPermissions are not changed")
+  end
+end  
+
+function Test:Precondition_registerAppAndSendSetAppIcon()
+  self.mobileSession = mobile_session.MobileSession(self, self.mobileConnection)
+  self.mobileSession.version = 4
+  self.mobileSession:StartService(7)
+  :Do(function()
+    registerApplication(self)
+    EXPECT_NOTIFICATION("OnHMIStatus", { systemContext = "MAIN", hmiLevel = "NONE", audioStreamingState = "NOT_AUDIBLE"})
+    :Do(function()
+     local cidPutFile = self.mobileSession:SendRPC("PutFile",
+      {
+        syncFileName = "iconFirstApp.png",
+        fileType = "GRAPHIC_PNG",
+        persistentFile = false,
+        systemFile = false
+      }, "files/icon.png")
+     EXPECT_RESPONSE(cidPutFile, { success = true, resultCode = "SUCCESS" })
+     :Do(function()
+     local cidSetAppIcon = self.mobileSession:SendRPC("SetAppIcon",{ syncFileName = "iconFirstApp.png" })
+     local pathToAppFolder = pathToAppFolderFunction(RAIParameters.appID)
+     EXPECT_HMICALL("UI.SetAppIcon",
+      {
+        syncFileName =
+          {
+            imageType = "DYNAMIC",
+            value = pathToAppFolder .. "iconFirstApp.png"
+          }
+       })
+    :Do(function(_,data)
+      self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS", {})
+    end)
+    EXPECT_RESPONSE(cidSetAppIcon, { resultCode = "SUCCESS", success = true })
+    end)
+    end)
+  end)
+end
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Test")
+
+function Test:Check_SDL_works_correctly_if_IconsFolder_has_no_permissions()
+  local status = true
+  local dirExistResult = commonFunctions:Directory_exist(folderNoPermissions)
+  if dirExistResult == true then
+    local applicationFileToCheck = folderNoPermissions .. "/" .. RAIParameters.appID
+    local applicationFileExistsResult = commonSteps:file_exists(applicationFileToCheck)
+    if applicationFileExistsResult ~= false then
+      commonFunctions:userPrint(31, RAIParameters.appID .. " icon is written to folder without permissions")
+      status = false
+    end
+    else 
+      commonFunctions:userPrint(31, "FolderWithoutPermissions folder does not exist in SDL bin folder" )
+      status = false
+    end
+    self:FailTestCase("SDL should not use the 'store-apps-icon' mechanism and store icon")
```

Comment:
@aarotar, Please note FailTestCase() function has to be invoked only in case if 'status' = false